motionplan/constraint.go

```diff
@@ -359,6 +367,28 @@ func NewCollisionConstraint(
 	return constraint, nil
 }
 
+// NewInteractionSpaceConstraint will determine if the given list of robot geometries are encompassed by the
+// given list of interactionSpaces. An interaction space is a geometry a robot must remain within while navigating a path.
+func NewInteractionSpaceConstraint(robotGeoms, interactionSpaces []spatial.Geometry) StateConstraint {
+	// determine if we would like to also return an ik.StateMetric to measure how much a robot geometry is encompasses by a given
+	// interaction space geometry
+	f := func(state *ik.State) bool {
+		for _, interactionSpace := range interactionSpaces {
```

Comment:
I feel like this might be cleaner if we tried to leverage the existing collision graph, but I havent thought this all the way through

Reply:
i will ponder on this as well 🤔 💭

Reply:
hmmmm, so in this function there is no existing collision graph, are you suggesting instead I create one?

Reply:
note: spoke offline with Ray about this.